packages/ODEs/src/scripting-tools.ts

```diff
@@ -594,26 +639,108 @@ function getMathArg(funcIdx: number): string {
   return (funcIdx > POW_IDX) ? '(x)' : '(x, y)';
 }
 
-/** Return custom output lines */
-function getCustomOutputLines(name: string, outputs: Map<string, Output>): string[] {
-  const res = [''];    
-  
+/** Return custom output lines: no expressions */
+function getCustomOutputLinesNoExpressions(name: string,
+  outputs: Map<string, Output>, toAddUpdateCol: boolean): string[] {
+  const res = [''];
+
   res.push(SCRIPT.CUSTOM_OUTPUT_COM);
   res.push(SCRIPT.CUSTOM_COLUMNS);
 
   outputs.forEach((val, key) => {
     if (!val.formula)
-      res.push(`${SCRIPT.SPACE2} DG.Column.fromFloat32Array('${val.caption}', ${DF_NAME}.col('${key}').getRawData()),`);
+      res.push(`${SCRIPT.SPACE2}${DF_NAME}.col('${key}'),`);
+  });
+
+  if (toAddUpdateCol)
+    res.push(`${SCRIPT.SPACE2}${DF_NAME}.col('${STAGE_COL_NAME}'),`);
+
+  res.push('];');
+
+  outputs.forEach((val, key) => {
+    if (!val.formula)
+      res.push(`${DF_NAME}.col('${key}').name = '${val.caption}';`);
   });
 
-  res.push('];');    
   res.push(`${DF_NAME} = DG.DataFrame.fromColumns(${COLUMNS});`);
   res.push(`${DF_NAME}.name = '${name}';`);
+  return res;
+} // getCustomOutputLinesNoExpressions
+
+/** Return custom output lines: with expressions */
+function getCustomOutputLinesWithExpressions(ivp: IVP): string[] {
+  const res = [''];
+
+  res.push(SCRIPT.CUSTOM_OUTPUT_COM);
+
+  // 1. Solution raw data
+  res.push(`const ${ivp.arg.name}RawData = ${DF_NAME}.col('${ivp.arg.name}').getRawData();`);
```

Comment:
push is a very expensive operations, maybe I would try to make an assign by index, but it is up to you

Reply:
Sure, but it requires some time to figure out how to implement this

Reply:
Currently, the largest is 193 (the FAE case)